### PR TITLE
Remove TransportClient auth/auth

### PR DESCRIPTION
**Pull request by jochenkressin**
### Description
With OpenSearch 2.0 the TransportClient is being removed. This means that there is no need for TransportClient auth/auth code anymore, or audit log categories that are connected with TansportLayer auth/auth. This PR removes the obsolete code and fixes some tests audit log unit tests.

### Issues Resolved
https://github.com/opensearch-project/security/issues/1578

Is this a backport? If so, please add backport PR # and/or commits #
No

### Testing
Ran the complete test suite, adapted tests that involved TransportClient auth/auth or obsolete audit log categories

### Check List
- [x] New functionality includes testing
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: src/test/java/org/opensearch/security/auditlog/impl/DisabledCategoriesTest.java [hunk 1]
```diff
@@ -195,7 +193,7 @@ protected boolean categoriesPresentInLog(String result, AuditCategory... categor
 	}
 
 	protected void logAll(AuditLog auditLog) {
-		//11 requests
+		//10 requests
```
Comment (peternied) [inline src/test/java/org/opensearch/security/auditlog/impl/DisabledCategoriesTest.java:196]: What does this comment mean?

**Review by peternied — DISMISSED** [review]

File: src/test/java/org/opensearch/security/auditlog/impl/DisabledCategoriesTest.java (hunk 1 above)
Comment (jochenkressin) [reply src/test/java/org/opensearch/security/auditlog/impl/DisabledCategoriesTest.java:196]: The comment was in there before, and I think it just states the obvious ;-) namely, we log/generate 10 events in this method. Agreed, it's unnecessary and I can remove it if needed.

Comment (peternied) [reply src/test/java/org/opensearch/security/auditlog/impl/DisabledCategoriesTest.java:196]: If you have an opportunity to remove lets do it, or we filter it out over time

**Review by peternied — DISMISSED** [review]

**Comment by cliu123** [comment]
@jochenkressin There're 10 test failures. Could you please take a look?

**Comment by jochenkressin** [comment]
There was indeed an issue where the privileges check for the injected user on the transport layer was not executed. The last commit fixes this. In the logs I found two other test failures that are IMHO unrelated to the changes:

**DlsTest#testDlsWithMinDocCountZeroAggregations**
This test seems to be flaky. When running it locally, it sometimes succeeds and sometimes fails

**BasicAuditlogTest#testRestMethod**
I cannot reproduce the test failure locally

**Comment by cliu123** [comment]
@jochenkressin Would you please rebase and resolved the conflicts?

**Comment by peternied** [comment]
> @jochenkressin Would you please rebase and resolved the conflicts?

Seeing as this change is very close to being merged, I'll take over making sure we can get it in for the 2.0.0 release.  Thanks @jochenkressin

**Review by peternied — APPROVED** [review]

**Comment by peternied** [comment]
@opensearch-project/security Could we get a second reviewer to take a look?

**Pull request merged**